signals.go

```diff
@@ -28,12 +28,20 @@ func newSignalHandler(enableSubreaper bool, notifySocket *notifySocket) *signalH
 	// ensure that we have a large buffer size so that we do not miss any signals
 	// in case we are not processing them fast enough.
 	s := make(chan os.Signal, signalBufferSize)
```

Comment:
nit: This channel could be created inside the goroutine instead, but it doesn't really matter.

Reply:
I pushed a fix manually.